add EKS Golang README with basic information and install instructions (#515)
aws/eks-distro-build-tooling: projects/golang/go/README.md

# EKS Supported Go Versions

> EKS supports a larger set of Golang version than upstream Golang. These versions are used to build Kubernetes and other Kubernetes ecosystem components used by EKS. Relevent upstream security fixes are backported to these Go versions. These patched versions are then built into RPMs and tested by building the relevent version of EKS Distro (e.g. Go 1.16 and Kubernetes 1.21) and executing the Kubernetes conformance tests.

Reviewer: ```suggestion
EKS supports a larger set of Golang version than upstream Golang. These versions are used to build Kubernetes and other Kubernetes ecosystem components used by EKS. Relevant upstream security fixes are backported to these Go versions. These patched versions are then built into RPMs and tested by building the relevant version of EKS Distro (e.g. Go 1.16 and Kubernetes 1.21) and executing the Kubernetes conformance tests.
```

## Upstream Patches
EKS Golang Versions are distributed as RPMs built from upstream Golang source for the given version with relevant security and utility patches, and their tests, backported.

> The patches applied to a given version of Golang are stored in the [EKS Distro Build Tooling Github repository](https://github.com/aws/eks-distro-build-tooling/tree/main/projects/golang/go), alongside the build system for EKS Golang. For example, the patches associated with EKS Go v1.16 can be found here https://github.com/aws/eks-distro-build-tooling/tree/main/projects/golang/go/1.15/patches. 

Author: ```suggestion
The patches applied to a given version of Golang are stored in the [EKS Distro Build Tooling Github repository](https://github.com/aws/eks-distro-build-tooling/tree/main/projects/golang/go), alongside the build system for EKS Golang. For example, you can see the patches associated with EKS Go v1.16 [here](https://github.com/aws/eks-distro-build-tooling/tree/main/projects/golang/go/1.16/patches). 
```

EKS currently supports the following Golang versions:
- Go v1.15.15
- Go v1.16.15

> EKS Go plans to suport all relevnt versions of Go (e.g. those used by a supported Kubernetes version or other Kubernetes ecosystem component) in the near future.

Reviewer: ```suggestion
EKS Go plans to support all relevant versions of Go (e.g. those used by a supported Kubernetes version or other Kubernetes ecosystem component) in the near future.
```
